src/app/modules/app/app.js

```diff
@@ -123,18 +139,106 @@
             });
-            $scope.topTags = tagService.getTags();
+
+                //This is the global list of all categories of tags
+                //The tags are separated out by these categories on the sidebar of the browse view
+                //id is the usage in the tag (e.g. "region.MyRegion") and displayName is the header
+                $scope.tagCategories = [
+                    {id: 'topic', displayName: 'Topics'},
+                    {id: 'region', displayName: 'Regions'},
+                    {id: 'level', displayName: 'Reading Levels'}
+                ];
+
+                //This is the object which will hold all of the tag names
+                $scope.tags = {topic: {top: [], other: []}};
+
+                tagService.getTags().then(function(tags) {
+                    //Get the names out of the tags; we don't care about the other properties
+                    var tagNames = tags.map(function(item) {
+                        return item.name;
+                    });
+
+                    //This is the number of tags that will be shown in each category of tag before the (n more...)
+                    var numberOfTopTags = 100;
+
+                    var i, j, cat;
+                    var categoryRegex = {};
+
+                    //Set up objects and regexes to save processing time
+                    for(i = 0; i < $scope.tagCategories.length; i++) {
+                        cat = $scope.tagCategories[i].id;
+                        $scope.tags[cat] = {top: [], other: []};
+                        categoryRegex[cat] = new RegExp('^' + cat + '\\.');
+                    }
+
+                    //Loop through tags
+                    for(i = 0; i < tagNames.length; i++) {
+                        var isPartOfCategory = false;
+
+                        //Check if tag belongs to a category
+                        for(j = 0; j < $scope.tagCategories.length; j++) {
+                            cat = $scope.tagCategories[j].id;
+                            if(categoryRegex[cat].test(tagNames[i])) {
+                                isPartOfCategory = true;
+
+                                //If we have more room in the top list, add to top list; otherwise, add to other list
+                                if($scope.tags[cat].top.length < numberOfTopTags) {
+                                    $scope.tags[cat].top.push(tagNames[i]);
+                                }
+                                else {
+                                    $scope.tags[cat].other.push(tagNames[i]);
+                                }
+                                break;
+                            }
+                        }
+                        if(!isPartOfCategory) {
+                            //Default is the first-listed category
+                            cat = $scope.tagCategories[0].id;
+
+                            //If we have more room in the top list, add to top list; otherwise, add to other list
```

Comment:
About six lines duplicated here. If you just set cat to $scope.tagCategories[0].id before the tagCategories loop, couldn't you remove the test above, and remove the isPartOfCategory flag, and unconditionally do this next bit once you have settled the category?